odoo/tools/mail.py

```diff
@@ -595,3 +596,30 @@ def formataddr(pair, charset='utf-8'):
                 name=email_addr_escapes_re.sub(r'\\\g<0>', name),
                 addr=address)
     return address
+
+def localpartify(email_name):
+    """
+    Turn any ascii string into a rfc2822 compliant local-part.
+
+    >>> 'john.doe, MyCompany Inc.'
+    'john.doe-MyCompany-Inc'
+    """
+
+    email_name.encode('ascii')  # would require SMTP-UTF8 otherwise
+
+    # https://www.jochentopf.com/email/chars.html
+    safechars = set(
+          string.ascii_lowercase
+        + string.ascii_uppercase
+        + string.digits
+        + '+-_.'
+    )
```

Comment:
This has no reason to be a local variable. Also might be faster with a regex and `re.sub`?